### PR TITLE
Improve DistributedQueryRunner.Builder

### PR DESCRIPTION
**Pull request by ssheikin**
## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: testing/trino-faulttolerant-tests/src/test/java/io/trino/faulttolerant/iceberg/TestIcebergParquetFaultTolerantExecutionConnectorTest.java [hunk 1]
```diff
@@ -37,12 +37,12 @@ protected IcebergQueryRunner.Builder createQueryRunnerBuilder()
         this.minioStorage = new MinioStorage("test-exchange-spooling-" + randomNameSuffix());
         minioStorage.start();
 
-        IcebergQueryRunner.Builder builder = super.createQueryRunnerBuilder();
-        getExtraProperties().forEach(builder::addExtraProperty);
-        builder.setAdditionalSetup(runner -> {
-            runner.installPlugin(new FileSystemExchangePlugin());
-            runner.loadExchangeManager("filesystem", getExchangeManagerProperties(minioStorage));
-        });
+        IcebergQueryRunner.Builder builder = super.createQueryRunnerBuilder()
```
Comment (losipiuk) [inline testing/trino-faulttolerant-tests/src/test/java/io/trino/faulttolerant/iceberg/TestIcebergParquetFaultTolerantExecutionConnectorTest.java:40]: nit `builder` variable not needed any more. inline

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java [hunk 2]
```diff
@@ -616,7 +616,7 @@ private static void closeUnchecked(AutoCloseable closeable)
     {
         private Session defaultSession;
         private int nodeCount = 3;
-        private Map<String, String> extraProperties = new HashMap<>();
+        private Map<String, String> extraProperties = ImmutableMap.of();
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:619]: I do not get this change. What was wrong with previous appraoch?

Comment (ssheikin) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:619]: HashMap.put/putAll could potentially silently overwrite existing value. This is not a case with ImmutableMap

Comment (losipiuk) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:617]: Oh I see. Just do `putIfAbsent` and assert it returns `null`. Code will be much simpler.

**Review by wendigo — APPROVED** [review]

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java (hunk 2 above)
Comment (ssheikin) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:617]: Despite 
>  Code will be much simpler.

This is a step backward from Immutable collections, which relaxes the code style - opens a possibility to override properties without intention. I'd prefer to left it as it is.

Comment (losipiuk) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:617]: I do not see it as such. It is not that ImmutableMap prevents you from overriding properties,  but the logic in `addProperties`/`addProperty`. 
Yet, whatever I will not fight over it.

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
@hashhar opinion here?

File: testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java (hunk 2 above)
Comment (hashhar) [reply testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java:617]: I agree with @losipiuk. Even if you have ImmutableMap you're still using a builder to construct it in those methods and anyone can override there if needed.

The safety comes from the fact that the only way to set these properties is via `addProperties/addProperty` and as long as those methods prevent overriding there's no possibility of overriding happening by mistake.

I also don't think it's worth arguing over even though the code looks much simpler and self-documenting with the change applied.

**Pull request merged**